src/omg-cli/omg/commands/commands.go

```diff
@@ -75,12 +75,16 @@ func selectedTiles(logger *log.Logger, config *config.EnvConfig) []tiles.TileIns
 	return result
 }
 
-type step func() error
+type step struct {
```

Comment:
An alternative to this would be to write a wrapper function. That way no matter where the code was used it would always print e.g.

```
func wrapStep(name string, s step) step {
  return func() error {
    fmt.Printf("Running step: %s\n", name)

    err := s()
    if err != nil {
      return fmt.Errorf("failed running step %s, error: %v", name, err)
    }
    return nil
  }
}
```

Composable wrapper functions could also ensure some actions are only done once, others have appropriate backoffs, and/or have retries built into the function itself to make the whole thing more declarative.

Reply:
That would be assuming the wrapper function is always used.

In this case, I don't think there's a reason why a step shouldn't have it's name printed and it's error message in the same format as the others. 

I agree that composite functions would be good for backoffs/retries, etc.